polaris-service/src/main/java/org/apache/polaris/service/catalog/BasePolarisCatalog.java

```diff
@@ -1873,6 +1878,51 @@ private boolean sendNotificationForTableLike(
     if (notificationType == NotificationType.DROP) {
       return dropTableLike(PolarisEntitySubType.TABLE, tableIdentifier, Map.of(), false /* purge */)
           .isSuccess();
+    } else if (notificationType == NotificationType.VALIDATE) {
+      // In this mode we don't want to make any mutations, so we won't auto-create non-existing
+      // parent namespaces. This means when we want to validate allowedLocations for the proposed
+      // table metadata location, we must independently find the deepest non-null parent namespace
+      // of the TableIdentifier, which may even be the base CatalogEntity if no parent namespaces
+      // actually exist yet. We can then extract the right StorageInfo entity via a normal call
+      // to findStorageInfoFromHierarchy.
+      PolarisResolvedPathWrapper resolvedStorageEntity = null;
+      Optional<PolarisEntity> storageInfoEntity = Optional.empty();
+      for (int i = tableIdentifier.namespace().length(); i >= 0; i--) {
```

Comment:
Do we still validate that this principle has privileges to create namespaces and table (if they don't exist)?

Reply:
Correct. This is validated in the updated [PolarisCatalogHandlerWrapperAuthzTest.java](https://github.com/apache/polaris/pull/321/files#diff-e24f0ec3325e14f90aa79137c1bf533824f1a1bcaa6b9f821f3a4ed60f771fa5)